### PR TITLE
Map new GameRule methods and EditGameRulesScreen

### PR DESCRIPTION
**Pull request by i509VCB**

### CODE REVIEW

File: mappings/net/minecraft/client/gui/screen/world/EditGameRulesScreen.mapping [hunk 1]
```diff
@@ -0,0 +1,54 @@
+CLASS net/minecraft/class_5235 net/minecraft/client/gui/screen/world/EditGameRulesScreen
+	FIELD field_24293 ruleSaveConsumer Ljava/util/function/Consumer;
```
Comment (liach) [inline mappings/net/minecraft/client/gui/screen/world/EditGameRulesScreen.mapping]:
```suggestion
	FIELD field_24293 ruleSaver Ljava/util/function/Consumer;
```

File: mappings/net/minecraft/client/gui/screen/world/EditGameRulesScreen.mapping [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	FIELD field_24294 ruleListWidget Lnet/minecraft/class_5235$class_5241;
+	FIELD field_24295 invalidRuleWidgets Ljava/util/Set;
+	FIELD field_24296 doneButton Lnet/minecraft/class_4185;
+	FIELD field_24297 tooltipDescription Ljava/util/List;
```
Comment (liach) [inline mappings/net/minecraft/client/gui/screen/world/EditGameRulesScreen.mapping]: Hmm, isn't a tooltip already a description? mind elaborate on this field?

File: mappings/net/minecraft/client/gui/screen/world/EditGameRulesScreen.mapping [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+			ARG 2 value
+	CLASS class_5237 RuleCategoryWidget
+		FIELD field_24306 name Lnet/minecraft/class_2561;
+		METHOD <init> (Lnet/minecraft/class_5235;Lnet/minecraft/class_2561;)V
+			ARG 2 text
+	CLASS class_5238 RuleWidgetFactory
+		METHOD create (Lnet/minecraft/class_2561;Ljava/util/List;Ljava/lang/String;Lnet/minecraft/class_1928$class_4315;)Lnet/minecraft/class_5235$class_5240;
+			ARG 1 name
+			ARG 2 description
+			ARG 4 rule
```
Comment (liach) [inline mappings/net/minecraft/client/gui/screen/world/EditGameRulesScreen.mapping:37]:
```suggestion
			ARG 3 ruleName
			ARG 4 rule
```
(assuming arg 3 is the rule name)

File: mappings/net/minecraft/client/gui/widget/EntryListWidget.mapping [hunk 4]
```diff
@@ -94,12 +94,14 @@ CLASS net/minecraft/class_350 net/minecraft/client/gui/widget/EntryListWidget
...
-			ARG 2 y
-			ARG 3 x
+			ARG 1 matrices
+			ARG 2 x
+			ARG 3 y
 			ARG 4 width
 			ARG 5 height
 			ARG 6 mouseX
 			ARG 7 mouseY
+			ARG 10 delta
```
Comment (liach) [inline mappings/net/minecraft/client/gui/widget/EntryListWidget.mapping]: `tickDelta`?

Comment (i509VCB) [reply mappings/net/minecraft/client/gui/widget/EntryListWidget.mapping]: I've seen both thrown around, would be nice if we standardized on a name

File: mappings/net/minecraft/client/gui/screen/world/EditGameRulesScreen.mapping (hunk 2 above)
Comment (i509VCB) [reply mappings/net/minecraft/client/gui/screen/world/EditGameRulesScreen.mapping]: Hmm yeah Its the tooltip

File: mappings/net/minecraft/client/gui/widget/EntryListWidget.mapping (hunk 4 above)
Comment (liach) [reply mappings/net/minecraft/client/gui/widget/EntryListWidget.mapping]: If it is a fraction between 0.0 and 1.0 for lerping a value between the last tick and present tick value, then it's `tickDelta`. If it is not tick related then just `delta`

**Review by liach — APPROVED** [review]

**Review by YanisBft — APPROVED** [review]

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**